test/e2e/upgrade/service/service.go

```diff
@@ -276,5 +276,9 @@ func TestReachableHTTPWithMinSuccessCount(host string, port int, successCount in
 		consecutiveSuccessCnt = 0
 		return false, nil // caller can retry
 	})
-	framework.ExpectNoError(err)
+	if err == wait.ErrWaitTimeout {
+		framework.Failf("service %s load balancer at %s:%d did not have %d consecutive, successful HTTP responses in %s", serviceName, host, port, successCount, timeout)
+	} else {
+		framework.ExpectNoError(err)
```

Comment:
You can actually fold these together - add the fmt.Sprintf at the end as the first optional param